### ISSUE TITLE
Dedup: Python backend frame adapters duplicate to_numpy/from_numpy/stack_coords across 5 modules (~110 LoC)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#525]
## Summary

The Python backend array adapters in `gamfit/` duplicate the same three primitives — *array → contiguous float64 numpy*, *numpy → backend tensor*, and *stack 1-D coordinate arrays into `(B, d)` with shape/length validation* — across five modules. `gamfit/_frame_torch.py` and `gamfit/torch/_coerce.py` are essentially bit-for-bit copies, and `gamfit/_basis_protocol.py` re-implements the coordinate-stacking validators instead of calling the frame adapters. **~110 LoC of duplication.**

(Note: the related "basis evaluation front doors" claim — `_basis_eval.py` / `torch/_basis.py` / `_basis_protocol.py` — was investigated and found **NOT** to be duplication; those are intentional, distinct layers, **except** that `_basis_protocol.py`'s coordinate stacking is the duplication captured here.)

## Locations

**`to_numpy_f64` (detach → CPU → cast f64 → contiguous → numpy)**
- `gamfit/_frame_torch.py:25-49`
- `gamfit/torch/_coerce.py:20-43` ← near-identical to the above
- `gamfit/_frame_jax.py:26-36`
- `gamfit/_frame_numpy.py:21-26`

**`from_numpy_like` (numpy → backend tensor, optional device/dtype match)**
- `gamfit/_frame_torch.py:52-64`
- `gamfit/torch/_coerce.py:58-70` ← lines 66-70 identical to `_frame_torch.py:59-63`

**`stack_coords` (validate each 1-D coord, length-match, stack to `(B,d)`)**
- `gamfit/_frame_torch.py:67-91`
- `gamfit/_frame_jax.py:56-79`
- `gamfit/_frame_numpy.py:29-52`
- `gamfit/_basis_protocol.py:159-178` (`_stack_coords`, torch)
- `gamfit/_basis_protocol.py:181-203` (`_stack_coords_numpy`)
- `gamfit/_basis_protocol.py:206-229` (`_stack_coords_jax`)

The validation loop (len > 0; per-coord coerce; assert 1-D; assert equal length; cast float; stack on axis/dim 1) and even the **error message text** are near-identical across all six stackers (e.g. `_frame_torch.py:78-86` vs `_basis_protocol.py:168-176`).

## Duplicated vs legitimately backend-specific

**Duplicated (should be shared):**
- The detach/cast/contiguous sequence (torch and jax versions ~95% identical).
- The coordinate length/shape validation loop (identical across all backends).

**Legitimately backend-specific (keep):**
- Device-placement policy (torch forces CPU; jax respects default-device policy).
- `_tc.float64` version-compat wrapper in `torch/_coerce.py`.
- `to_numpy_uintp` (torch-only).
- `wrap_value_grad` VJP routing (backend-specific autograd).

## Proposed fix

Make `_frame.py` + `_frame_<backend>.py` the single adapter layer with a small protocol:

```python
class BackendOps(Protocol):
    def to_numpy_f64(self, x): ...
    def from_numpy_like(self, arr, ref=None): ...
    def stack_coords(self, coords): ...
    def wrap_value_grad(self, ...): ...
```

- Factor the shared validation loop into a pure-numpy helper that each backend's `stack_coords` calls with its own `coerce` callable.
- `gamfit/torch/_coerce.py` delegates to `_frame_torch.py` (delete its copies).
- `gamfit/_basis_protocol.py` calls the backend adapters' `stack_coords` instead of carrying `_stack_coords`/`_stack_coords_numpy`/`_stack_coords_jax`.

## Impact
- Removes ~110 LoC of duplicated conversion/validation.
- Single place to fix coordinate-shape semantics; today a fix must be applied in up to six spots.

---
*Verified by reading all five modules and diffing the functions. Confidence: high for the frame-adapter duplication. Cluster 6 of a 13-cluster semantic-duplication sweep.*

**Comment by SauersML** [comment]
Fixed in `f7137873c`. New `_frame_shared.py::stack_coords_generic` owns the empty/1-D/equal-length validation + error text once; numpy/jax/torch `stack_coords` delegate. `torch/_coerce.py` `to_numpy_f64`/`from_numpy_like` now delegate to `_frame_torch` (kept torch-only `to_numpy_uintp` + non-tensor-ref guard). `_basis_protocol.py` deleted its three private stackers and routes through the adapters (lazy imports preserve the "import gamfit never imports torch/jax" guarantee). CI verifies.
